pom.xml

```diff
@@ -180,7 +180,8 @@ under the License.
     <resolverVersion>2.0.0-alpha-5</resolverVersion>
     <securityDispatcherVersion>2.0</securityDispatcherVersion>
     <sisuVersion>0.9.0.M2</sisuVersion>
-    <slf4jVersion>1.7.36</slf4jVersion>
+    <slf4jVersion>2.0.9</slf4jVersion>
+    <slf4jSimpleVersion>1.7.36</slf4jSimpleVersion>
```

Comment:
Won't this cause a warning with SLF4J?

Reply:
Not really, at least not at runtime.  The reason is that we're not really using slf4j simple directly, but we use our own provider.  However, that one cannot be easily upgrade to 2.x as it's missing two key features: the ability to override the string used for displaying the log level, and the ability to more nicely format exceptions.
Those have been removed from 2.x.

Reply:
See https://jira.qos.ch/browse/SLF4J-394

Reply:
@michael-o  What warning are you thinking of? As long as you furnish the appropriate implementation of SLF4JServiceProvider, SLF4J 2.0 should be satisfied. 

@gnodet I would recommend to use another package instead of org.slf4j.impl for example org.slf4j.maven in order JMPS/Jigsaw conflicts.

Reply:
> Not really, at least not at runtime. The reason is that we're not really using slf4j simple directly, but we use our own provider which extends slf4j-simple. However, that one cannot be easily upgrade to 2.x as it's missing two key features: the ability to override the string used for displaying the log level, and the ability to more nicely format exceptions.
> Those have been removed from 2.x.

@gnodet I am not aware of this, in any case it was not done intentionally.

Reply:
Changing the package name is not trivial, as we extend `SimpleLogger` from slf4j 1.x which has a package protected method.

@ceki do you think this commit https://github.com/qos-ch/slf4j/commit/509c3444ac7 could be back ported to slf4j 2.x ?

Reply:
> Changing the package name is not trivial, as we extend `SimpleLogger` from slf4j 1.x which has a package protected constructor.
> 
> @ceki do you think this commit [qos-ch/slf4j@509c3444ac7](https://github.com/qos-ch/slf4j/commit/509c3444ac7) could be backported to slf4j 2.x ?

Tags say it is...?!

Reply:
> @michael-o What warning are you thinking of? As long as you furnish the appropriate implementation of SLF4JServiceProvider, SLF4J 2.0 should be satisfied.

SLF4J used to warning when version of API and Impl weren't identical...

Reply:
@gnodet Yes, of course. It looks like only the `renderLevel` method was missing. I added it back in commit 7029a8b7fda 
in the main branch (i.e the 2.0 branch). This should allow you to move to slf4j-simple version 2.0. Let me know if other changes are needed.

Reply:
@gnodet Also changed the access modifier of the constructor for SimpleLogger to protected. See [commit fe42ecb5f](https://github.com/qos-ch/slf4j/commit/fe42ecb5f74cdda9927fe0889229199a08d23ec8)

Reply:
@michael-o 

> SLF4J used to warning when version of API and Impl weren't identical...

Good point. Fortunately, the requested API version is returned by the **provider** in the `getResquestedVersion` method. So you should be good.

Reply:
@ceki there's a few more protected methods needed in order to be reusable from a different package, see https://github.com/qos-ch/slf4j/pull/385